SU2_CFD/src/solvers/CTransLMSolver.cpp

```diff
@@ -37,480 +40,461 @@
+      nLineLets = Jacobian.BuildLineletPreconditioner(geometry, config);
+      if (rank == MASTER_NODE) cout << "Compute linelet structure. " << nLineLets << " elements in each line (average)." << endl;
+    }
 
     LinSysSol.Initialize(nPoint, nPointDomain, nVar, 0.0);
     LinSysRes.Initialize(nPoint, nPointDomain, nVar, 0.0);
+    System.SetxIsZero(true);
 
-    /*--- Jacobians and vector structures for implicit computations ---*/
-    if (config->GetKind_TimeIntScheme_Turb() == EULER_IMPLICIT) {
+    if (ReducerStrategy)
+      EdgeFluxes.Initialize(geometry->GetnEdge(), geometry->GetnEdge(), nVar, nullptr);
 
-      /*--- Point to point Jacobians ---*/
-      Jacobian_i = new su2double* [nVar];
-      Jacobian_j = new su2double* [nVar];
-      for (iVar = 0; iVar < nVar; iVar++) {
-        Jacobian_i[iVar] = new su2double [nVar];
-        Jacobian_j[iVar] = new su2double [nVar];
-      }
-      /*--- Initialization of the structure of the whole Jacobian ---*/
-      Jacobian.Initialize(nPoint, nPointDomain, nVar, nVar, true, geometry, config);
+    /*--- Initialize the BGS residuals in multizone problems. ---*/
+    if (multizone){
+      Residual_BGS.resize(nVar,0.0);
+      Residual_Max_BGS.resize(nVar,0.0);
+      Point_Max_BGS.resize(nVar,0);
+      Point_Max_Coord_BGS.resize(nVar,nDim) = su2double(0.0);
+    }
 
-      if (config->GetKind_Linear_Solver_Prec() == LINELET) {
-        nLineLets = Jacobian.BuildLineletPreconditioner(geometry, config);
-        if (rank == MASTER_NODE) cout << "Compute linelet structure. " << nLineLets << " elements in each line (average)." << endl;
-      }
+  }
 
-    }
+  /*--- Initialize lower and upper limits---*/
+  lowerlimit[0] = 1.0e-4;
+  upperlimit[0] = 5.0;
+
+  lowerlimit[1] = 1.0e-4;
+  upperlimit[1] = 1.0e15;
+
+  /*--- Far-field flow state quantities and initialization. ---*/
+  su2double Intensity;
+
+  Intensity = config->GetTurbulenceIntensity_FreeStream()*100.0;
+
+  su2double Intermittency_Inf  = 1.0;
+  su2double ReThetaT_Inf = 100.0;
+  su2double f_lambda = 0.0, lambda = 0.0;
 
-    /*--- Read farfield conditions from config ---*/
-    Intermittency_Inf = config->GetIntermittency_FreeStream();
-    tu_Inf            = 100.0 * config->GetTurbulenceIntensity_FreeStream();
+  f_lambda = 1. - (-12.986*lambda - 123.66*lambda*lambda -
+                         405.689*lambda*lambda*lambda)*exp(-pow(2./3*Intensity,1.5));
 
-    /*-- Initialize REth from correlation --*/
-    if (tu_Inf <= 1.3) {
-      REth_Inf = (1173.51-589.428*tu_Inf+0.2196/(tu_Inf*tu_Inf));
-    } else {
-      REth_Inf = 331.5*pow(tu_Inf-0.5658,-0.671);
+
+  /*--- Momentum thickness Reynolds number, initialized from freestream turbulent intensity*/
+  if (Intensity <= 1.3) {
+    if(Intensity >=0.027) {
+      ReThetaT_Inf = (1173.51-589.428*Intensity+0.2196/(Intensity*Intensity)) * f_lambda;
+    }
+    else {
+      ReThetaT_Inf = (1173.51-589.428*Intensity+0.2196/(0.27*0.27))*f_lambda;
```

Comment:
For consistency: does this have to be 0.027*0.027 as this forces the minimum value of the Intensity of 0.027?

Reply:
Also, there shouldn't be any f_lambda multiplication for the freestream conditions.